plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/DeallocateVmVspCommand.java

```diff
@@ -83,9 +92,130 @@ public String getVmUuid() {
         return _vmUuid;
     }
 
+    public boolean isExpungingState() {
+        return _isExpungingState;
+    }
+
+    public static class Builder implements CmdBuilder<DeallocateVmVspCommand> {
+        private String _networkUuid;
+        private String _nicFromDdUuid;
+        private String _nicMacAddress;
+        private String _nicIp4Address;
+        private boolean _isL3Network;
+        private boolean _isSharedNetwork;
+        private String _vpcUuid;
+        private String _networksDomainUuid;
+        private String _vmInstanceName;
+        private String _vmUuid;
+        private boolean _isExpungingState;
+
+        public Builder networkUuid(String networkUuid) {
+            this._networkUuid = networkUuid;
+            return this;
+        }
+
+        public Builder nicFromDbUuid(String nicFromDbUuid) {
+            this._nicFromDdUuid = nicFromDbUuid;
+            return this;
+        }
+
+        public Builder nicMacAddress(String nicMacAddress) {
+            this._nicMacAddress = nicMacAddress;
+            return this;
+        }
+
+        public Builder nicIp4Address(String nicIp4Address) {
+            this._nicIp4Address = nicIp4Address;
+            return this;
+        }
+
+        public Builder isL3Network(boolean isL3Network) {
+            this._isL3Network = isL3Network;
+            return this;
+        }
+
+        public Builder isSharedNetwork(boolean isSharedNetwork) {
+            this._isSharedNetwork = isSharedNetwork;
+            return this;
+        }
+
+        public Builder vpcUuid(String vpcUuid) {
+            this._vpcUuid = vpcUuid;
+            return this;
+        }
+
+        public Builder networksDomainUuid(String networksDomainUuid) {
+            this._networksDomainUuid = networksDomainUuid;
+            return this;
+        }
+
+        public Builder vmInstanceName(String vmInstanceName) {
+            this._vmInstanceName = vmInstanceName;
+            return this;
+        }
+
+        public Builder vmUuid(String vmUuid) {
+            this._vmUuid = vmUuid;
+            return this;
+        }
+
+        public Builder isExpungingState(boolean isExpungingState) {
+            this._isExpungingState = isExpungingState;
+            return this;
+        }
+
+        @Override
+        public DeallocateVmVspCommand build() {
+            return new DeallocateVmVspCommand(_networkUuid,_nicFromDdUuid, _nicMacAddress, _nicIp4Address, _isL3Network, _isSharedNetwork, _vpcUuid,
+                    _networksDomainUuid, _vmInstanceName, _vmUuid, _isExpungingState);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
This ``equals`` implementation should include attributes from the ``Command`` super class.